src/CommitmentsV2/SFA.DAS.CommitmentsV2.Api.UnitTests/Controllers/CohortControllerTests.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.CommitmentsV2.Api.Controllers;
+using SFA.DAS.CommitmentsV2.Api.Types;
+using SFA.DAS.CommitmentsV2.Api.Types.Requests;
+using SFA.DAS.CommitmentsV2.Application.Commands.AddCohort;
+using SFA.DAS.CommitmentsV2.Mapping;
+
+namespace SFA.DAS.CommitmentsV2.Api.UnitTests.Controllers
+{
+    [TestFixture]
+    public class CohortControllerTests
+    {
+        [Test]
+        public Task CreateController_ValidRequest_ShouldReturnAnOkResult()
+        {
+            return new CohortControllerTestFixtures()
+                        .AssertOkayResponse();
+        }
+
+        [Test]
+        public Task CreateController_ValidRequest_ShouldReturnExpectedExpectedResponseObject()
+        {
+            return new CohortControllerTestFixtures()
+                        .AssertExpectedResponseType();
+        }
+
+        [Test]
+        public Task CreateController_ValidRequest_ShouldReturnExpectedCohortId()
+        {
+            return new CohortControllerTestFixtures()
+                .AssertResponseIsValid(response => 
+                    Assert.AreEqual(CohortControllerTestFixtures.CohortId, response.Id));
+        }
+
+        [Test]
+        public Task CreateController_ValidRequest_ShouldReturnExpectedReference()
+        {
+            return new CohortControllerTestFixtures()
+                .AssertResponseIsValid(response =>
+                    Assert.AreEqual(CohortControllerTestFixtures.Reference, response.Reference));
+        }
+
+        [Test]
+        public Task CreateController_ValidRequest_ShouldReturnExpectedApprenticeshipId()
+        {
+            return new CohortControllerTestFixtures()
+                .AssertResponseIsValid(response => 
+                    Assert.AreEqual(CohortControllerTestFixtures.DraftApprenticeshipId, response.DraftApprenticeshipId));
+        }
+    }
+
+    public class CohortControllerTestFixtures
+    {
+        public CohortControllerTestFixtures()
+        {
+            MediatorMock = new Mock<IMediator>();
+            MapperMock = new Mock<IMapper<CreateCohortRequest, AddCohortCommand>>();
+        }
+
+        private Mock<IMediator> MediatorMock { get; }
+        private IMediator Mediator => MediatorMock.Object;
+
+        private Mock<IMapper<CreateCohortRequest, AddCohortCommand>> MapperMock { get; }
+        private IMapper<CreateCohortRequest, AddCohortCommand> Mapper => MapperMock.Object;
+
+        public const long CohortId = 123;
+        public const string Reference = "ABC123";
+        public const long DraftApprenticeshipId = 456;
+
```

Comment:
I see what you mean but I think the following order is conventional:

privates fields, 
contstructor, 
public properties, 
public methods, 
private methods


These are _public_ so I would expected to see them _after_ the constructor.
I think they look odd because they are public _constants_, not properties so are public fields and therefore not common and not normally seen.

Reply:
I've made the change here: https://github.com/SkillsFundingAgency/das-commitments/pull/463/commits/4da1b6344a19436b3e9b3ba569f386245564a881